### PR TITLE
New develop version

### PR DESCRIPTION
**Pull request by qunxyz**
supported aead cipher and udp tunnel

### CODE REVIEW

**Comment by arthurkiller** [comment]
would you like to fix the CI?

**Comment by qunxyz** [comment]
yes. I thought it should be your job, sorry for that. will be done in the next few days

**Comment by arthurkiller** [comment]
@qunxyz I mean your PR should pass the CI

**Comment by arthurkiller** [comment]
This really a big PR, and will be pending(just like mine PR and others) after I get some idea about how to do the reconstruction on this repo

**Comment by arthurkiller** [comment]
Maybe a developer branch

**Comment by qunxyz** [comment]
got it. and I also changed some code for cow project, I will make a pr for cow till you merged this branch

**Comment by arthurkiller** [comment]
About cow, u'd contact the author for more help

**Comment by qunxyz** [comment]
I tried doing some minor change to support aead, but then I realized that seems impossible. so it is this branch. anyway, maybe you would liked this branch

File: cmd/shadowsocks-httpget/httpget.go
Comment (arthurkiller) [inline cmd/shadowsocks-httpget/httpget.go]: I don't want this be depend on. cause the shadowsocks-go should provide a function full server and client and the MOST IMPORTANT is a easy use lib for other users

**Review by arthurkiller — CHANGES_REQUESTED** [review]

**Comment by qunxyz** [comment]
never mind cow, though it's the reason I developed this branch. I just waiting. take your time

File: cmd/shadowsocks-httpget/httpget.go
Comment (arthurkiller) [reply cmd/shadowsocks-httpget/httpget.go]: about locally support AEAD, I have do this in my PR. But it will not merge yet.

**Comment by arthurkiller** [comment]
yep, and I'm sorry cause this PR will not be merged in this way. I will make a plan how we do the reconstruct on this repo, and welcome you to join.

File: cmd/shadowsocks-httpget/httpget.go
Comment (qunxyz) [reply cmd/shadowsocks-httpget/httpget.go]: I don't want that neither. it is a mistake, I didn't noticed that line before. about AEAD, I have to do some jobs to make cow keep working for me, cause my ss provider had upgraded ss cipher to chacha20-ietf-poly1305. according to shadowsocks-go original functions' structure, I couldn't found another better way to support it. maybe yours is better, for now my version is working quite well. btw, maybe this pr should be abandoned, I would make a new pr in the next few days

Comment (arthurkiller) [reply cmd/shadowsocks-httpget/httpget.go]: THX for your contribute and maybe we can get another way to contribute

Comment (qunxyz) [reply cmd/shadowsocks-httpget/httpget.go]: parts of this branch implement for dealing Stream and AEAD cipher might not call elegant, maybe even ugly, I hate it a lot. but I am not familar with golang, maybe someone would fixed or rewrote that in future.

**Comment by qunxyz** [comment]
never mind. but anyway, I would made a new pr once done the parts of passing CI. merge or not, it's all your call, I would accepted that. I just want to share it, and I appreciate you creating this project.

**Comment by arthurkiller** [comment]
@qunxyz dude, would you like to fix this merge request and push to another branch I will create for your reconstruction?

**Review by arthurkiller — APPROVED** [review]

**Comment by qunxyz** [comment]
so sorry, @arthurkiller. I forgot it, will be done this weekend

**Comment by arthurkiller** [comment]
Great! And Now I will create a new branch called `dev-qunxyz` for your merge request

**Comment by qunxyz** [comment]
maybe after two days? I should made a new pr on Sunday afternoon

**Comment by arthurkiller** [comment]
never mind

**Comment by arthurkiller** [comment]
solve conflicts and change the dependencies into shadowsocks-go

**Comment by qunxyz** [comment]
oh, man, you already done that. thanks very much, I really appreciate it.
then I don't need to make a new pr for it. just let you know, I fixed it too

**Comment by arthurkiller** [comment]
LGTM

**Comment by arthurkiller** [comment]
@qunxyz still get some conflicts, rebase the master

**Comment by arthurkiller** [comment]
still here?@qunxyz

**Comment by arthurkillerMB** [comment]
@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB

**Comment by arthurkillerMB** [comment]
@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB

**Comment by arthurkillerMB** [comment]
@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB

**Comment by arthurkillerMB** [comment]
@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB

**Comment by arthurkillerMB** [comment]
@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB

**Comment by arthurkillerMB** [comment]
@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB

**Comment by arthurkillerMB** [comment]
@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB

**Comment by arthurkillerMB** [comment]
@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB

**Comment by arthurkillerMB** [comment]
@arthurkiller 审核你MB帐户状态++你+arthurkillerMB+arthurkillerMB

**Comment by artificerpi** [comment]
Any updates on this PR?

**Pull request closed without merging**
